core/domain/question_services_test.py

```diff
@@ -30,91 +29,101 @@ class QuestionServicesUnitTest(test_utils.GenericTestBase):
+        self.question = self.save_new_question(
+            self.question_id, self.editor_id,
+            self._create_valid_question_data('ABC'))
 
     def test_get_question_by_id(self):
-        question = question_domain.Question(
-            'dummy', self._create_valid_question_data('ABC'), 1, 'en')
-
-        question_id = question_services.add_question(self.owner_id, question)
-        question = question_services.get_question_by_id(question_id)
+        question = question_services.get_question_by_id(
+            self.question_id, strict=False)
 
-        self.assertEqual(question.question_id, question_id)
+        self.assertEqual(question.id, self.question_id)
 
     def test_get_questions_by_ids(self):
-        question = question_domain.Question(
-            'dummy', self._create_valid_question_data('ABC'), 1, 'en')
-
-        question1_id = question_services.add_question(
-            self.owner_id, question)
-        question = question_domain.Question(
-            'dummy2', self._create_valid_question_data('ABC'), 1, 'en')
-
-        question2_id = question_services.add_question(
-            self.owner_id, question)
+        question_id_2 = question_services.get_new_question_id()
+        self.save_new_question(
+            question_id_2, self.editor_id,
+            self._create_valid_question_data('DEF'))
         questions = question_services.get_questions_by_ids(
-            [question1_id, question2_id])
+            [self.question_id, question_id_2])
         self.assertEqual(len(questions), 2)
-        self.assertEqual(questions[0].question_id, question1_id)
-        self.assertEqual(questions[1].question_id, question2_id)
-
-    def test_add_question(self):
-        question_data = self._create_valid_question_data('ABC')
-        question_id = 'dummy'
-        question_data_schema_version = 1
-        language_code = 'en'
-        question = question_domain.Question(
-            question_id, question_data, question_data_schema_version,
-            language_code)
-
-        question_id = question_services.add_question(self.owner_id, question)
-        model = question_models.QuestionModel.get(question_id)
-
-        self.assertEqual(model.question_data, question_data)
-        self.assertEqual(
-            model.question_data_schema_version,
-            question_data_schema_version)
-        self.assertEqual(model.language_code, language_code)
+        self.assertEqual(questions[0].id, self.question_id)
+        self.assertEqual(questions[1].id, question_id_2)
 
     def test_delete_question(self):
-        question = question_domain.Question(
-            'dummy', self._create_valid_question_data('ABC'), 1, 'en')
-
-        question_id = question_services.add_question(self.owner_id, question)
-        question_services.delete_question(
-            self.owner_id, question_id)
+        question_services.delete_question(self.editor_id, self.question_id)
```

Comment:
Also test the case where the question doesn't exist.

In general don't just write one example per test. Cover the whole "space" of inputs. Ditto below/elsewhere.

Reply:
Done, added tests for get question rights as well.